pyfar/io/io.py

```diff
@@ -297,3 +228,210 @@ def write(filename, compress=False, **objs):
 
     with open(filename, 'wb') as f:
         f.write(zip_buffer.getvalue())
+
+
+def read_audio(filename, dtype='float64', **kwargs):
+    """
+    Import an audio file as :py:class:`~pyfar.classes.audio.Signal` object.
+
+    Parameters
+    ----------
+    filename : string, Path
+        Input file.
+    dtype : {'float64', 'float32', 'int32', 'int16'}, optional
+        Data type of the returned signal, by default ``'float64'``.
+        Floating point audio data is typically in the range from
+        ``-1.0`` to ``1.0``.  Integer data is in the range from
+        ``-2**15`` to ``2**15-1`` for ``'int16'`` and from ``-2**31`` to
+        ``2**31-1`` for ``'int32'``.
+    **kwargs
+        Other keyword arguments to be passed to ``soundfile.read``. This is
+        needed, e.g, to read RAW audio files.
+
+
+    Returns
+    -------
+    signal : Signal
+        :py:class:`~pyfar.classes.audio.Signal` object containing the audio
+        data.
+
+    Notes
+    -----
+    * This function is based on ``soundfile.read``.
+    * Reading int values from a float file will *not* scale the data to
+      [-1.0, 1.0). If the file contains ``np.array([42.6], dtype='float32')``,
+      you will read ``np.array([43], dtype='int32')`` for ``dtype='int32'``.
+    """
+    data, sampling_rate = sf.read(
+        file=filename, dtype=dtype, always_2d=True, **kwargs)
+    signal = Signal(data.T, sampling_rate, domain='time', dtype=dtype)
+    return signal
+
+
+def write_audio(signal, filename, subtype=None, overwrite=True, **kwargs):
+    """
+    Write a :py:class:`~pyfar.classes.audio.Signal` object as a audio file to
+    disk.
+
+    Parameters
+    ----------
+    signal : Signal
+        Object to be written.
+    filename : string, Path
+        Output file. The format is determined from the file extension.
+        See :py:func:`available_formats` for all possible formats.
+    subtype : str, optional
+        The subtype of the sound file, the default value depends on the
+        selected `format` (see :py:func:`default_subtype`).
+        See :py:func:`available_subtypes` for all possible subtypes for
+        a given ``format``.
+    overwrite : bool
+        Select wether to overwrite the audio file, if it already exists.
+        The default is ``True``.
+    **kwargs
+        Other keyword arguments to be passed to ``soundfile.write``.
+
+    Notes
+    -----
+    * Signals are flattened before writing to disk (e.g. a signal with
+      ``cshape = (3, 2)`` will be written to disk as a six channel audio file).
+    * This function is based on ``soundfile.write``.
+    * Amplitudes larger than +/- 1 are clipped.
```

Comment:
This is not always true. Writing wav-files with float or double will not clip. Suggestion - find out when this is the case and raise a warning if data is actually clipped. We could than omit this note.

Reply:
As this is not straighforward for all implemented formats, I added the function ``write_audio_clipped``.